sdk/storage/AzureStorageBlob/Source/Operations/ContainersOperations.swift

```diff
@@ -0,0 +1,287 @@
+        )
+
+        // Construct and send request
+        let codingKeys = PagedCodingKeys(
+            items: "EnumerationResults.Containers",
+            continuationToken: "EnumerationResults.NextMarker",
+            xmlItemName: "Container"
+        )
+        let xmlMap = XMLMap(withPagedCodingKeys: codingKeys, innerType: ContainerItem.self)
+        let context = PipelineContext.of(keyValues: [
+            ContextKey.xmlMap.rawValue: xmlMap as AnyObject,
+            ContextKey.xmlErrorMap.rawValue: StorageError.xmlMap() as AnyObject
+        ])
+        context.add(cancellationToken: options?.cancellationToken, applying: client.options)
+        context.merge(with: options?.context)
+        guard let requestUrl = client.url(template: urlTemplate, params: params),
+            let request = try? HTTPRequest(method: .get, url: requestUrl, headers: params.headers) else {
+            client.options.logger.error("Failed to construct Http request")
+            return
+        }
+
+        client.request(request, context: context) { result, httpResponse in
+            let dispatchQueue = options?.dispatchQueue ?? self.client.commonOptions.dispatchQueue ?? DispatchQueue.main
+            switch result {
+            case let .success(data):
+                guard let data = data else {
+                    let noDataError = AzureError.client("Response data expected but not found.")
+                    dispatchQueue.async {
+                        completionHandler(.failure(noDataError), httpResponse)
+                    }
+                    return
+                }
+                do {
+                    let decoder = StorageJSONDecoder()
+                    let paged = try PagedCollection<ContainerItem>(
+                        client: self.client,
+                        request: request,
+                        context: context,
+                        data: data,
+                        codingKeys: codingKeys,
+                        decoder: decoder
+                    )
+                    dispatchQueue.async {
+                        completionHandler(.success(paged), httpResponse)
+                    }
+                } catch {
+                    dispatchQueue.async {
+                        completionHandler(.failure(AzureError.client("Decoding error.", error)), httpResponse)
+                    }
+                }
+            case let .failure(error):
+                dispatchQueue.async {
+                    completionHandler(.failure(error), httpResponse)
+                }
+            }
+        }
+    }
+
+    /// Create a storage container in a storage account.
+    /// - Parameters:
+    ///   - container: The container name to create.
+    ///   - options: A `CreateContainerOptions` object to control the create operation.
+    ///   - completionHandler: A completion handler that receives a dictionary of properties on
```

Comment:
It's not really a dictionary, we should word this better.

Reply:
Updated.